dali/kernels/imgproc/convolution/filter_gpu.cuh

```diff
@@ -0,0 +1,1231 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_CONVOLUTION_FILTER_GPU_CUH_
+#define DALI_KERNELS_IMGPROC_CONVOLUTION_FILTER_GPU_CUH_
```

Comment:
I'd recommend splitting this file into:
`filter_gpu.h`, containing the kernel class, parameters, etc
and
`filter_gpu_impl.cuh` with GPU kernels.
Depending on build times, we might even decide to instantiate just some input/output combinations in `dali_kernels` to avoid compiling this file twice - in kernel tests and operators.

Reply:
Done